### PR TITLE
chore: add sign and submit with passwoard

### PR DESCRIPTION
**Pull request by SorinC6**
task: 
- https://emurgo.atlassian.net/browse/YOMO-771
- https://emurgo.atlassian.net/browse/YOMO-792

### CODE REVIEW

**Comment by michalszorad** [comment]
This one is still WIP, right?

**Comment by SorinC6** [comment]
> This one is still WIP, right?

Hi, yes! but the flow related to the sign and submit with password is almost ready

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 1]
```diff
@@ -6,17 +7,47 @@ import {AmountItem} from '../../../../components/AmountItem/AmountItem'
...
+
+  const poolFee = Quantities.denominated(
+    `${Number(Object.values(unsignedTx?.fee))}`,
+    Number(wallet.primaryTokenInfo.decimals),
+  )
+
+  const feesInfo = [
+    {
+      label: strings.swapMinAdaTitle,
+      value: '2 ADA',
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: Hardcoded value

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 2]
```diff
@@ -6,17 +7,47 @@ import {AmountItem} from '../../../../components/AmountItem/AmountItem'
...
+      info: strings.swapMinAda,
+    },
+    {
+      label: strings.swapMinReceivedTitle,
+      value: '?', // TODO add real value
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx:42]: Hardcoded value

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx [hunk 3]
```diff
@@ -6,17 +7,47 @@ import {AmountItem} from '../../../../components/AmountItem/AmountItem'
...
   const wallet = useSelectedWallet()
+  const {createOrder, unsignedTx} = useSwap()
+  const {amounts} = createOrder
+
+  const buyTokenInfo = useTokenInfo({wallet, tokenId: amounts.buy.tokenId})
+  const tokenToBuyName = buyTokenInfo.ticker ?? buyTokenInfo.name
+  const label = `${Quantities.format(amounts.buy.quantity, buyTokenInfo.decimals ?? 0)} ${tokenToBuyName}`
+
+  const poolFee = Quantities.denominated(
+    `${Number(Object.values(unsignedTx?.fee))}`,
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: Is this computed correctly?

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/LedgerTransportSwitch.tsx [hunk 4]
```diff
@@ -0,0 +1,86 @@
...
+import {spacing} from '../../../../theme'
+import {HARDWARE_WALLETS, useLedgerPermissions} from '../../../../yoroi-wallets/hw'
+import {useStrings} from '../../common/strings'
+
+type Props = {
+  onSelectUSB: () => void
+  onSelectBLE: () => void
+}
+
+const useIsUsbSupported = () => {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/LedgerTransportSwitch.tsx]: I see we have a similar hook in apps/wallet-mobile/src/HW/LedgerTransportSwitchModal/LedgerTransportSwitchModal.tsx. Can we reuse it or move to shared folder?

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxWithPassword.tsx [hunk 5]
```diff
@@ -0,0 +1,76 @@
...
+        testID="swapButton"
+        shelleyTheme
+        title={strings.sign}
+        onPress={() => signAndSubmitTx({unsignedTx, password: spendingPassword})}
+      />
+
+      {isLoading && (
+        <View
+          style={{
+            position: 'absolute',
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxWithPassword.tsx]: Let's move inline styles to `styles`

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxWithHW.tsx [hunk 6]
```diff
@@ -0,0 +1,89 @@
...
+type Props = {
+  wallet: YoroiWallet
+  unsignedTx: YoroiUnsignedTx
+  onCancel?: () => void
+  onSuccess: () => void
+}
+
+type TransportType = 'USB' | 'BLE'
+
+export const ConfirmTxWithHW = (props: Props) => {
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxWithHW.tsx:24]: Can we re-use apps/wallet-mobile/src/Dashboard/WithdrawStakingRewards/ConfirmTx/ConfirmTxWithHW.tsx ?

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx (hunk 2 above)
Comment (SorinC6) [reply apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx:42]: for this one we are still missing the formula -> `min receiver`

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxWithHW.tsx (hunk 6 above)
Comment (SorinC6) [reply apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/ConfirmTxWithHW.tsx:24]: here 
```
    <Confirm {...props} transport={transportType} />
```
from `ConfirmTxWithHW ` is a bit different and maybe will change the component more 🤔

File: apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx (hunk 3 above)
Comment (stackchain) [reply apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: fee is BN -> use `asQuantity` helper

Comment (michalszorad) [reply apps/wallet-mobile/src/features/Swap/useCases/ConfirmTxScreen/TransactionSummary.tsx]: I noticed the following is also used 

```
formatTokenWithText(
        Amounts.getAmount(unsignedTx.fee, wallet.primaryToken.identifier).quantity,
        wallet.primaryToken,
      )
```

**Comment by jorbuedo** [comment]
@SorinC6 What's the status? What help do you need? Is it about the backend error?

**Comment by SorinC6** [comment]
> @SorinC6 What's the status? What help do you need? Is it about the backend error?

Hi @jorbuedo  the PR is ready now, Krimo helped to fix the issue on submitTx

File: apps/wallet-mobile/src/features/Swap/common/mocks.ts [hunk 7]
```diff
@@ -0,0 +1,46 @@
...
+        fee: '0.05',
+        lastUpdate: '2023-09-08 09:56:13',
+        lpToken: {
+          quantity: `${68917682}` as `${number}`,
+          tokenId: '0029cb7c88c7567b63d1a512c0ed626aa169688ec980730c0473b913.6c702083',
+        },
+        poolId: '0029cb7c88c7567b63d1a512c0ed626aa169688ec980730c0473b913.702083',
+        price: 0.0890390378168252,
+        provider: 'sundaeswap' as ProviderType,
+        tokenA: {quantity: `${20630071}` as `${number}`, tokenId: ''},
```
Comment (michalszorad) [inline apps/wallet-mobile/src/features/Swap/common/mocks.ts]: instead of `${number}` we could use `BalanceQuantity`

**Review by michalszorad — COMMENTED** [review]
Hello, I've noticed that when I try to place a limit order, the button is unresponsive. I've noticed the warning `[Error: Protocol needs to be one of [minswap, wingriders, sundaeswap, muesliswap, muesliswap_v2]]` in console, so maybe it's related. I've tried to Swap 1 ADA for `3,333,333,333` HOSKY at the limit price `0.0000000003`

**Comment by jorbuedo** [comment]
> Hello, I've noticed that when I try to place a limit order, the button is unresponsive. I've noticed the warning `[Error: Protocol needs to be one of [minswap, wingriders, sundaeswap, muesliswap, muesliswap_v2]]` in console, so maybe it's related. I've tried to Swap 1 ADA for `3,333,333,333` HOSKY at the limit price `0.0000000003`

Following yesterday's conversation. I don't think we have any protocol validation on the app. So I assume this error came from the backend?

**Comment by SorinC6** [comment]
> > Hello, I've noticed that when I try to place a limit order, the button is unresponsive. I've noticed the warning `[Error: Protocol needs to be one of [minswap, wingriders, sundaeswap, muesliswap, muesliswap_v2]]` in console, so maybe it's related. I've tried to Swap 1 ADA for `3,333,333,333` HOSKY at the limit price `0.0000000003`
> 
> Following yesterday's conversation. I don't think we have any protocol validation on the app. So I assume this error came from the backend?


yeah, seems like for `Spectrum` protocol - some of the pairs have this as default have this issues that the protocol is not supported but as Jordi said we don't have yet a validation. We could create a ticket for this to be solved in other PR later

File: apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts [hunk 8]
```diff
@@ -907,6 +909,7 @@ export const makeShelleyWallet = (constants: typeof MAINNET | typeof TESTNET) =>
     }
 
     async submitTransaction(signedTx: string) {
+      console.log('submitTransaction signedTx', signedTx)
```
Comment (stackchain) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts]:
```suggestion
      console.log('submitTransaction signedTx', signedTx)
```

File: apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts [hunk 9]
```diff
@@ -621,7 +621,7 @@ export const makeShelleyWallet = (constants: typeof MAINNET | typeof TESTNET) =>
       }
     }
 
-    async signTx(unsignedTx: YoroiUnsignedTx, decryptedMasterKey: string) {
+    async signTx(unsignedTx: YoroiUnsignedTx, decryptedMasterKey: string, datum?: any) {
```
Comment (stackchain) [inline apps/wallet-mobile/src/yoroi-wallets/cardano/shelley/ShelleyWallet.ts]: update the type (ShelleyWallet + ByronWallet)

File: apps/wallet-mobile/src/yoroi-wallets/hooks/index.ts [hunk 10]
```diff
@@ -420,6 +421,7 @@ export const useSignWithHwAndSubmitTx = (
       ...options?.signTx,
       onSuccess: (signedTx, args, context) => {
         options?.signTx?.onSuccess?.(signedTx, args, context)
+        console.log('useSignWithHwAndSubmitTx', {signedTx, args, context})
```
Comment (stackchain) [inline apps/wallet-mobile/src/yoroi-wallets/hooks/index.ts]:
```suggestion
        console.log('useSignWithHwAndSubmitTx', {signedTx, args, context})
```

File: packages/swap/src/translators/reactjs.tsx
Comment (stackchain) [inline packages/swap/src/translators/reactjs.tsx:1]: I've updated it, can you revert, and reapply later if needed?

**Pull request merged**